src/GridParser.ts

```diff
@@ -0,0 +1,63 @@
+import { GridNotFoundErr, headerNotFoundErr, invalidDataTypeErr, invalidValueInColumnErr } from "./constants";
+import { ColumnMetadata, dataTypes, Grid, TableData } from "./types";
+
+export default class GridParser {
+    public parseToTableData(initialDataGrid: Grid): TableData | undefined {
```

Comment:
This class is not static, is there a reason why we don't just send the Grid in the factory and own it? 
or maybe we don't need the factory and we can make it all static methods?